src/common/utils.ts

```diff
@@ -169,13 +169,54 @@ export const toggleDebugMode = (toggleBtn: HTMLElement): void => {
   toggleBtn.innerText = "Dev";
 };
 
-export const hideElement = (
-  isHide: boolean = false,
-  element: HTMLElement
-) => {
+export const hideElement = (isHide: boolean = false, element: HTMLElement) => {
   if (isHide) {
     element.classList.remove("show");
   } else {
     element.classList.add("show");
   }
-};
\ No newline at end of file
+};
+
+export function applyFontToElement(
+  element: HTMLElement | null,
+  fontOptions: { 
+    customFonts: Record<string, string>, 
+    defaultFont: string, 
+    lang: string 
+  },
+  fontBasePath: string
+): void {
+  const { customFonts, defaultFont, lang } = fontOptions;
+  
+  // Determine the correct font name based on the language
+  const fontName = customFonts[lang] || defaultFont;
+  const fontPath = `${fontBasePath}${fontName}.ttf`;
+
+  // Ensure fontName and fontPath exist
+  if (!fontName || !fontPath) {
+    return;
+  }
+
+  // Generate a unique ID for the font style element
+  const styleId = `font-${fontName.replace(/\s+/g, '-').toLowerCase()}`;
+
+  // Check if the font style has already been added to the document head
+  if (!document.getElementById(styleId)) {
+    const style = document.createElement("style");
+    style.id = styleId;
+    style.textContent = `
+      @font-face {
+        font-family: '${fontName}';
+        src: url('${fontPath}') format('truetype');
+        font-weight: normal;
+        font-style: normal;
+      }
+    `;
+    document.head.appendChild(style);
```

Comment:
This function is being used by feedback-text-effects and start-scene.ts, so are we creating two elements?
Do we always append to the document.head every time feedback text loaded?

Reply:
@bernhardccodev no. i created a checker for that if the font has been already added to the document head then this code will not execute https://github.com/curiouslearning/FeedTheMonsterJS/blob/82eb47cd558468fb7c00897842c8b5cae0a50ad2/src/common/utils.ts#L204-L216

Reply:
why not include the check together with 
if (!fontName || !fontPath) {
    return;
  }

Reply:
@bernhardccodev done here.

Reply:
Let me rephrase my question: If this function needs to be triggered only once then why are me calling this in two separate files?

I just can't understand because we are checking if the font styles are already preloaded but feedback text will still call this method. Yes you have a condition that prevents it from running the condition again but that is why I am having a hard time due to it was already set but one file may still call this even though it has already set the font. Maybe we can optimize how this is being used so it will only be called once and no need for multiple files to trigger the font setting?

Reply:
@bernhardccodev i made an improvements here. i know load and apply the custom font in the feedthemonster.js file and removed the method i used on both files. this way it reduces the code.